### ISSUE TITLE
Exclusive processing during scan

### ISSUE THREAD

**Issue by tesshucom** [tesshucom/jpsonic#1475]
Although it is unlikely will have a fatal problem right away, there are some processes that do not want to run in parallel with the scanning process. To a lesser extent, but has a negative impact on performance. And, the impact will be greater under the condition of multi-scan.

<details>
<summary>Processes that do not want to run in parallel with the scanning process</summary>

These are originally features whose number of executions should be controlled, even if not scanning.

 - The fatal feature is **Fast access mode**. This has been removed already so there is no problem. `#1452` 
   - This issue deals with other risks and low priority issues. 
 - Restraints will be added  to some process, **to prevent processing from starting during the scan**.
   - Some of the features related to podcasts, tags, and indexes will be constrained. But impact on user operation is small.
     - In most cases, the user will not be aware of the constraint.. Because their features are not used often on during the scan.
   - Those features do **not require very strict exclusion**. It is not considered now if these features are done just before the scan. 
     - For example, you can start scanning immediately after you manually start downloading a very large number of podcasts.
     - In general, scanning is longer and more expensive than downloading Podacast.
     - This is the current specification(Even in Legacy, there is no such control and it is considered not so fatal.). Changes may be made if the podcast is reimplemented in more detail in the future.
   - `#1438` - [Research](https://github.com/tesshucom/jpsonic/issues/1438#issuecomment-1094297101)

</details>

It's not fatally essential, but it's a type of feature improvements that contributes more to stability. **Robust optimization**.

<details>
<summary>Fixes</summary>

#### [Batch(podcast updates)]

 - Update interval n(hour) is a setting item by the user
 - [Current task spec] Executed 5 minutes after startup, task registration n hours later
   - [x] Change spec for auto podcast updates
     - It is determined whether scanning is in progress 5 minutes after startup.
       - Podcasts will be updated if not scanning
       - If scanning is in progress, the same judgment will be repeated every 5 minutes until the scan is completed

#### [REST]

 - [x] Disable podcast create/download/refresh during scanning
   - If executed, be notified with a general error
 - ~~Fix not to reflesh when using getIndexes during scan~~ No fix required. Always refresh is false.

#### [WEB]

 - [x] Disable podcast create/download/refresh during scanning
   - If executed from a page that has already been displayed, no processing will be performed and a redirect will be performed.
 - [x] Fix not to reflesh when getting Index during scan
     - ~~SubsonicRESTController#getIndexes~~ No fix required. Always refresh is false.
     - ~~HomeController#get~~ No fix required. Always refresh is false.
     - ~~TopController#get~~ No fix required. Already fixed(TopControllerTest$RefleshTest).
       - LeftController for Airsonic. In Jpsonic, Left was integrated into Top. At that time, the update logic was also fixed.
     - [x] IndexUpnpProcessor#refreshIndex
 - [x] Disable editting tags during scanning

</details>

### Usage by the user

You probably don't need to be too aware, but some features will not be available during the scan.

#### [Batch(podcast updates)]

Podcast updates will start automatically 5 minutes after launch. After that, it will be executed periodically (the execution interval can be changed on the setting page).

From this version, if this schedule conflicts with a scan, the schedule will be automatically staggered.

![image](https://user-images.githubusercontent.com/27724847/166413471-1a2236a6-d643-48dc-ab21-9a4692883c41.png)

#### [Subsonic API]

Creating/Downloading/Updating podcasts during scanning will not be available. If executed, you will be notified with an error.　(Download here means download from the host of podcasts to Jpsonic. It does not affect playback or download from Jpsonic to the app)

#### [WEB pages]

 - Adds restrictions to the podcast feature, similar to the Subsonic API
 - You will not be able to edit tags while scanning

#### [UPnP]

 - If the index is browsed while the scan is in progress, no directory scan will be performed to avoid duplicate processing. After the scan is finished, you can get the latest index by executing refresh in the client application.
